packages/server/postgres/migrations/1703031300000_addEmbeddingTables.ts

```diff
@@ -0,0 +1,107 @@
+import {Client} from 'pg'
+import getPgConfig from '../getPgConfig'
+import {r} from 'rethinkdb-ts'
+import connectRethinkDB from '../../database/connectRethinkDB'
+
+export async function up() {
+  const client = new Client(getPgConfig())
+  await client.connect()
+  await client.query(`
+  DO $$
+  BEGIN
+      IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'EmbeddingsObjectTypeEnum') THEN
+          EXECUTE 'CREATE TYPE "EmbeddingsObjectTypeEnum" AS ENUM (''retrospectiveDiscussionTopic'')';
+      END IF;
+      IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'EmbeddingsStateEnum') THEN
+          EXECUTE 'CREATE TYPE "EmbeddingsStateEnum" AS ENUM (''queued'', ''embedding'', ''failed'')';
+      END IF;
+
+      EXECUTE 'CREATE TABLE IF NOT EXISTS "EmbeddingsJobQueue" (
+          "id" SERIAL PRIMARY KEY,
+          "objectType" "EmbeddingsObjectTypeEnum" NOT NULL,
+          "refId" VARCHAR(100) NOT NULL,
+          "model" TEXT NOT NULL,
+          UNIQUE("objectType", "refId", "model"),
+          "state" "EmbeddingsStateEnum" DEFAULT ''queued'' NOT NULL,
+          "stateMessage" TEXT,
+          "stateUpdatedAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL
+      )';
+
+      EXECUTE 'CREATE INDEX IF NOT EXISTS "idx_EmbeddingsJobQueue_objectType" ON "EmbeddingsJobQueue"("objectType")';
+      EXECUTE 'CREATE INDEX IF NOT EXISTS "idx_EmbeddingsJobQueue_refId" ON "EmbeddingsJobQueue"("refId")';
+      EXECUTE 'CREATE INDEX IF NOT EXISTS "idx_EmbeddingsJobQueue_state" ON "EmbeddingsJobQueue"("state")';
+      EXECUTE 'CREATE INDEX IF NOT EXISTS "idx_EmbeddingsJobQueue_stateUpdatedAt" ON "EmbeddingsJobQueue"("stateUpdatedAt")';
+
+      EXECUTE 'CREATE TABLE IF NOT EXISTS "EmbeddingsIndex" (
+        "id" SERIAL PRIMARY KEY,
+        "objectType" "EmbeddingsObjectTypeEnum" NOT NULL,
+        "refId" VARCHAR(100),
+        UNIQUE("objectType", "refId"),
+        "refDateTime" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        "models" TEXT[],
+        "teamId" VARCHAR(100) NOT NULL,
+        "orgId" VARCHAR(100) NOT NULL,
```

Comment:
-1 this gets really tricky because a team can change orgs. either we need to ditch `orgId` or we need to augment the `moveTeamToOrg` mutation to handle this table, too.

Reply:
blerg...this is true. I would be ok dropping this an just joining through the table instead. I will make this update.